platform/linux-generic/test/validation/api/ml/ml_linux.c

```diff
@@ -0,0 +1,1259 @@
+	ret = odp_ml_model_info(global.ml_model, &ml_info);
+	CU_ASSERT_EQUAL(ret, 0);
+	CU_ASSERT_STRING_EQUAL(ml_info.name, MODEL_NAME);
+	CU_ASSERT_EQUAL(ml_info.model_version, 1);
+	CU_ASSERT_EQUAL(ml_info.num_inputs, NUM_INPUTS);
+	CU_ASSERT_EQUAL(ml_info.num_outputs, NUM_OUTPUTS);
+
+	num_ret = odp_ml_model_input_info(global.ml_model, input_info, NUM_INPUTS);
+	CU_ASSERT_EQUAL(num_ret, NUM_INPUTS);
+	CU_ASSERT_STRING_EQUAL(input_info[0].name, "x");
+	CU_ASSERT_EQUAL(input_info[0].shape.num_dim, 1);
+	CU_ASSERT_EQUAL(input_info[0].shape.dim[0], 1);
+	CU_ASSERT_EQUAL((int)input_info[0].data_type, ODP_ML_DATA_TYPE_INT32);
+
+	/* When num is 0, return normally, and input_info is ignored */
+	num_ret = odp_ml_model_input_info(global.ml_model, input_info, 0);
+	CU_ASSERT_EQUAL(num_ret, NUM_INPUTS);
+
+	/* When num is bigger than actual number of inputs, extra input_info is left untouched */
+	input_info[1].data_type = (odp_ml_data_type_t)-1;
+	num_ret = odp_ml_model_input_info(global.ml_model, input_info, NUM_INPUTS + 1);
+	CU_ASSERT_EQUAL(num_ret, NUM_INPUTS);
+	CU_ASSERT_STRING_EQUAL(input_info[0].name, "x");
+	CU_ASSERT_EQUAL(input_info[0].shape.num_dim, 1);
+	CU_ASSERT_EQUAL(input_info[0].shape.dim[0], 1);
+	CU_ASSERT_EQUAL((int)input_info[0].data_type, ODP_ML_DATA_TYPE_INT32);
+	/* input_info[1] is left untouched */
+	CU_ASSERT_EQUAL(input_info[1].data_type, (odp_ml_data_type_t)-1);
+
+	num_ret = odp_ml_model_output_info(global.ml_model, output_info, NUM_OUTPUTS);
+	CU_ASSERT_EQUAL(num_ret, NUM_OUTPUTS);
+	CU_ASSERT_STRING_EQUAL(output_info[0].name, "y");
+	CU_ASSERT_EQUAL(output_info[0].shape.num_dim, 1);
+	CU_ASSERT_EQUAL(output_info[0].shape.dim[0], 1);
+	CU_ASSERT_EQUAL((int)output_info[0].data_type, ODP_ML_DATA_TYPE_INT32);
+
+	/* When num is 0, return normally, and input_info is ignored */
+	num_ret = odp_ml_model_output_info(global.ml_model, output_info, 0);
+	CU_ASSERT_EQUAL(num_ret, NUM_OUTPUTS);
+
+	/* When num is bigger than actual number of inputs, extra output_info is left untouched */
+	num_ret = odp_ml_model_output_info(global.ml_model, output_info, NUM_OUTPUTS + 1);
+	output_info[1].shape.num_dim = 98876;
+	CU_ASSERT_EQUAL(num_ret, NUM_OUTPUTS);
+	CU_ASSERT_STRING_EQUAL(output_info[0].name, "y");
+	CU_ASSERT_EQUAL(output_info[0].shape.num_dim, 1);
+	CU_ASSERT_EQUAL(output_info[0].shape.dim[0], 1);
+	CU_ASSERT_EQUAL((int)output_info[0].data_type, ODP_ML_DATA_TYPE_INT32);
+	/* output_info[1] is left untouched */
+	CU_ASSERT_EQUAL(output_info[1].shape.num_dim, 98876);
+}
+
+static void test_ml_model_load(void)
+{
+	int ret;
+	odp_ml_model_t test_model;
+	odp_ml_load_result_t result;
+
+	test_model = odp_ml_model_create(NULL, &global.model_param);
+	CU_ASSERT_NOT_EQUAL_FATAL(test_model, ODP_ML_MODEL_INVALID);
+
+	ret = odp_ml_model_load(test_model, &result);
+	CU_ASSERT_EQUAL(ret, 0);
+	CU_ASSERT_EQUAL(result.error_code, 0);
+
+	ret = odp_ml_model_load(test_model, NULL);
+	CU_ASSERT(ret);
```

Comment:
Validation test should not try to load again an already loaded model. It's not guaranteed that implementation will catch that.